extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -45,14 +144,33 @@ require(
 				return $.proxy(func, this);
 			},
 
-			getBaseUrl: function() {
+			getBaseUrl: function () {
 				return mw.config.get('wgOnSiteNotificationsApiUrl');
 			}
 
 		};
 
+		function compileMustache() {
+			loader({
+				type: loader.MULTI,
+				resources: {
+					mustache: 'extensions/wikia/DesignSystem/services/templates/DesignSystemGlobalNavigationOnSiteNotifications.mustache',
+					scripts: 'design_system_on_site_notifications_mustache_js'
+				}
+			}).done(function (assets) {
+				loader.processScript(assets.scripts);
+				OnSiteNotifications.init(assets.mustache[0]);
+			});
+		}
+
+		function getSafely(obj, path) {
+			return path.split(".").reduce(function (acc, key) {
+				return (typeof acc == "undefined" || acc === null) ? acc : acc[key];
```

Comment:
What if `typeof acc === 'string'`? Shouldn't we return the `acc` too, then? How about checking if it's an array or object?